### DISCUSSION TITLE
What is the meaning of a 404 return code?

### DISCUSSION THREAD

**Discussion by wsw70** [zinclabs/zinc#224] [Q&A] [answered]
I can send sucessfully the following request to create or update a document:

```
PUT https://zinc.example.com/api/hello/_doc/87378675-b4f6-451f-80e2-44876ee525ec HTTP/1.1
content-type: application/json
Authorization: Basic admin:v2gD5jwQDk

{
  "name": "sdsdsd",
  "age": 23,
  "zeze": true,
  "yuhu": 38
}
```

I am trying to send the same request from JavaScript, but I get a `404` code. Other requests such as a `/_search` are OK (from JS) so this is not an authentication or similar issue.

In order to debug this, **I would like to understand what exactly the `404` code means** (an issue with the URL? with the body?)

**Comment by prabhatsharma** [comment, marked as answer]
404 means you are hitting an API endpoint that does not exist. It is an issue with the URL that you are passing. You might want to check the URL again. In order to test your code you can try it on playground too.
```
https://playground.dev.zincsearch.com/ 
uid: admin
pwd: Complexpass#123
```

post in your sample code if you need help.

**Reply by wsw70** [reply]
> 404 means you are hitting an API endpoint that does not exist. It is an issue with the URL that you are passing. You might want to check the URL again. In order to test your code you can try it on playground too.

Thank you. I will create some code for that (the one I use is in too many pieces), would you have an idea what in the following URL can be wrong? 

```
{"level":"error","method":"PUT","code":404,"took":0,"time":"2022-06-22T19:54:52Z","message":"/api/_doc/hello/be58b66e-9fe0-4a67-b27f-6669bddda8f0"}
{"level":"info","method":"PUT","code":404,"took":0,"time":"2022-06-22T19:54:52Z","message":"/api/_doc/hello/be58b66e-9fe0-4a67-b27f-6669bddda8f0"}
```

AHHHH - I just realized (when adding the logs for successful calls)  that I inversed the index and `_doc`. Thanks for the hint!
